Xamarin.Build.AsyncTask/WaitForBackgroundTasks.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Threading;
+using System.Collections.Generic;
+using TPL = System.Threading.Tasks;
+
+namespace Xamarin.Build
+{
+    /// <summary>
+    /// This Task will wait for all the background tasks to complete.
+    /// You can control which task categories to wait on by using the
+    /// `Categories` property.
+    /// </summary>
+    public class WaitForBackgroundTasks : AsyncTask
+    {
+        internal static readonly string [] DefaultCategories = { BackgroundTaskManager.DefaultCategory };
+        /// <summary>
+        /// A list of background task categories to wait on.
+        /// Once all the tasks in each of these categories have Completed
+        /// the task will exit.
+        /// By default it will wait on the `default` category.
+        /// </summary>
+        public string [] Categories { get; set; }
+
+        /// <summary>
+        /// The error code to use should any of the background tasks fail.
+        /// this will default to XAT0000
+        /// </summary>
+        public string ErrorCode { get; set; } = "XAT0000";
+
+        public override bool Execute ()
+        {
```

Comment:
Does this method need the `Yield`, `try-finally`, `Reacquire` calls like we do here?

https://github.com/xamarin/xamarin-android/blob/fd64aaff618f5f818831bdef844876e138fec7a0/src/Xamarin.Android.Build.Tasks/Tasks/AndroidTask.cs#L47-L60